### PR TITLE
Prevent chpl-language-server from providing type inserts for loops

### PR DESCRIPTION
**Pull request by jabraham17**
This fixes an issue where `chpl-language-server` would add insertable type inlays for loops, even though thats not valid syntax. If a user tried to insert the type inlay, they would then get a syntax error. This PR prevents users from making that mistake by removing the insertable inlay. The inlay still shows, it is just not insertable.

Prior to this PR, users woild be able to insert the type as shown below
Screenshot 2024-05-31 at 3 11 56 PM

Inserting this type creates the following error
Screenshot 2024-05-31 at 3 13 11 PM

This PR prevents this situation from occurring all together.

Tested locally with `make test-cls`

[Reviewed by @DanilaFe]

### CODE REVIEW

**Comment by DanilaFe** [comment]
A picture of the bug-to-be-fixed would help the PR OP.

File: tools/chpl-language-server/test/util/utils.py [hunk 1]
```diff
@@ -522,25 +522,42 @@ async def check_type_inlay_hints(
...
 ) -> typing.List[InlayHint]:
     """
     Helper method for `check_inlay_hints`. Adds the `: ` prefix.
 
-    Also checks that inlays are insertable
+    `expected_inlays` is a list of tuples. The tuples can be either of length 2
+    or 3. If the tuple is of length 2, the first element is the expected
+    position and the second element is the expected text of the type. If the
+    tuple is of length 3, then the first two elements are the same. The third
+    element is a boolean that indicates whether the inlay hint is insertable.
```
Comment (DanilaFe) [inline tools/chpl-language-server/test/util/utils.py:538]: suggest mentioning that length-2 tuples imply insertable

File: tools/chpl-language-server/test/util/utils.py [hunk 2]
```diff
@@ -522,25 +522,42 @@ async def check_type_inlay_hints(
...
 
     # Check that the inlays are insertable
     for expected, actual in zip(inlays, actual_inlays):
-        # the list of inlay text edits should have one element and have the
-        # same text/range as the inlay
-        assert actual.text_edits is not None
-        assert len(actual.text_edits) == 1
-        assert actual.text_edits[0].range.start == expected[0]
-        assert actual.text_edits[0].new_text == expected[1]
+        print(expected, actual)
```
Comment (DanilaFe) [inline tools/chpl-language-server/test/util/utils.py:550]: Is this deliberate or accidental?

**Review by DanilaFe — APPROVED** [review]

File: tools/chpl-language-server/test/util/utils.py (hunk 2 above)
Comment (jabraham17) [reply tools/chpl-language-server/test/util/utils.py:550]: accidental, removed

**Review by DanilaFe — APPROVED** [review]

**Pull request merged**